projects/robots/gctronic/e-puck/plugins/remote_controls/e-puck_wifi/Wrapper.cpp

```diff
@@ -175,38 +200,52 @@ int Wrapper::robotStep(int step) {
       break;
     if (command == 0 && header == 3 && all_read)
       break;
-
-    log("received %d bytes\n", n);
     if (n == -1) {
       log("failed to received header\n");
       exit(1);
     }
     if (n > 0) {
       // log("received %d header\n", header);
       if (header == 0x01) {  // camera image
-        if (cCommunication->receive((char *)image, IMAGE_SIZE, true) == -1) {
+#ifdef LOG_COMMUNICATION_TIME
+        start = clock();
+#endif
+        n = cCommunication->receive((char *)image, IMAGE_SIZE, true);
+        if (n == -1) {
           log("Failed to receive camera image.\n");
           exit(1);
         }
-        log("Got image (location = %p size = %d)\n", image, IMAGE_SIZE);
-#ifndef NDEBUG
+#ifdef LOG_COMMUNICATION_TIME
+        end = clock();
+        cpu_time_used = (end - start) * 1000 / CLOCKS_PER_SEC;
+        log("Took %d ms to receive image (%d bytes)\n", cpu_time_used, n);
+#endif
+#ifdef SAVE_CAMERA_IMAGES
         // save the image as BMP file to debug
-        static int i = 0;
+        static int image_counter = 0;
         char filename[32];
-        sprintf(filename, "image%03d.png", i);
+        sprintf(filename, "image%03d.png", image_counter);
         unsigned char *rgb888 = (unsigned char *)malloc(160 * 120 * 3);
         rgb565_to_brg888(image, rgb888, 160, 120);
         save_bmp_image(filename, rgb888, 160, 120);
         free(rgb888);
-        i++;
+        image_counter++;
 #endif
         got_camera_image++;
       } else if (header == 0x02) {  // sensor data
-        if (cCommunication->receive((char *)sensor_data, SENSOR_DATA_SIZE, true) == -1) {
+#ifdef LOG_COMMUNICATION_TIME
+        start = clock();
+#endif
+        n = cCommunication->receive((char *)sensor_data, SENSOR_DATA_SIZE, true);
+        if (n == -1) {
```

Comment:
The receive() return code in case of failure is explicitly `0` not `1`.

```suggestion
        if (n == 0) {
```